api/nakadi-event-bus-api.yaml

```diff
@@ -606,22 +488,388 @@ paths:
           description: Partition id
           type: string
           required: true
-        - name: X-Flow-Id
+        - $ref: '#/parameters/XFlowId'
+      responses:
+        '200':
+          description: OK
+          schema:
+            $ref: '#/definitions/Partition'
+        '401':
+          description: Client is not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+
+  /subscriptions:
+    post:
+      tags:
+        - managed-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: |
+        This endpoint creates a subscription for EventTypes. The subscription is needed to be able to
+        consume events from EventTypes in a high level way when Nakadi stores the offsets and manages the
+        rebalancing of consuming clients.
+        The subscription is identified by its key parameters (owning_application, event_types, consumer_group). If
+        this endpoint is invoked several times with the same key subscription properties in body - the
+        subscription will be created only once and for all other calls it will just return the subscription
+        that was already created.
+      parameters:
+        - name: subscription
+          in: body
+          description: Subscription to create
+          schema:
+            $ref: '#/definitions/Subscription'
+          required: true
+      responses:
+        '200':
+          description: |
+            Subscription for such parameters already exists. Returns subscription object that already
+            existed.
+          schema:
+            $ref: '#/definitions/Subscription'
+        '201':
+          description: Subscription was successfuly created. Returns subscription object that was created.
+          schema:
+            $ref: '#/definitions/Subscription'
```

Comment:
Hmm, I'm not sure about the requirements by the guidelines here ...

[Must: Use Meaningful HTTP Status Codes](http://zalando.github.io/restful-api-guidelines/http/Http.html#must-use-meaningful-http-status-codes) says to "Always set the Location header" for 201 (i.e. when you created something).

[Content-Location](http://zalando.github.io/restful-api-guidelines/headers/CommonHeaders.html#contentlocation) details when/how to use Content-Location (here: when a non-GET request returns something which has its own address).